src/history/hs_rec.c

```diff
@@ -633,6 +635,9 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
     if (ret == 0 && insert_cnt > 0)
         __hs_verbose_cache_stats(session, btree);
 
+    if (insert_cnt > 0)
```

Comment:
The stat that gets updated based on this boolean is "page written requiring history store records". I am not sure if this function fails in itself, ie ret != 0, we would actually write a page. But the page write could fail even after successfully writing the whole history.

Maybe add a comment that says something like we return true if we have written anything to history, even if the function fails, to make it clear that we intentionally return true on failure too.